base/boot.jl

```diff
@@ -269,9 +269,9 @@ Void() = nothing
 
 immutable VecElement{T}
     value::T
-    VecElement{T}(value::T) where T = new(value) # disable converting constructor in Core
+    VecElement{T}(value::T) where {T} = new(value) # disable converting constructor in Core
 end
-VecElement(arg::T) where T = VecElement{T}(arg)
+VecElement(arg::T) where {T} = VecElement{T}(arg)
```

Comment:
Best demonstrated by example. What would

```
Complex{Float64}(x::Float64) = ...
```

mean?

Also note the old syntax is very close to

```
Vec{T}(arg) = Vec{T}(arg)
```

which looks circular. The `{T}` on the left means something very different than the `{T}` on the right.